### PR TITLE
Add support for System.Boolean

### PR DESCRIPTION
**Pull request by timdeschryver**
# Info
This Pull Request is related to issue no. #77

# Changes
This change adds the following methods:
- `IsTrue`
- `IsFalse`
- `IsEqualTo`
- `Required`

### CODE REVIEW

File: src/Valit/ValitRuleBooleanExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+namespace Valit
+{
+  public static class ValitRuleBooleanExtensions
+  {
+    public static IValitRule<TObject, bool> IsTrue<TObject>(this IValitRule<TObject, bool> rule) where TObject : class
+        => rule.Satisfies(p => p);
+
+    public static IValitRule<TObject, bool?> IsTrue<TObject>(this IValitRule<TObject, bool?> rule) where TObject : class
```
Comment (GooRiOn) [inline src/Valit/ValitRuleBooleanExtensions.cs]: I think that this Satisfies also requires p.HasValue

File: src/Valit/ValitRuleBooleanExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        => rule.Satisfies(p => p == true);
+
+    public static IValitRule<TObject, bool> IsFalse<TObject>(this IValitRule<TObject, bool> rule) where TObject : class
+        => rule.Satisfies(p => !p);
+
+    public static IValitRule<TObject, bool?> IsFalse<TObject>(this IValitRule<TObject, bool?> rule) where TObject : class
+        => rule.Satisfies(p => p == false);
```
Comment (GooRiOn) [inline src/Valit/ValitRuleBooleanExtensions.cs]: Same here

**Review by GooRiOn — CHANGES_REQUESTED** [review]
First of all, thank you so much for helping with Valit! Your code LGTM, except two missing checks (HasValue) for nullables. Please take a look in a free time :)

File: src/Valit/ValitRuleBooleanExtensions.cs (hunk 1 above)
Comment (timdeschryver) [reply src/Valit/ValitRuleBooleanExtensions.cs]: 👍 changed it.

File: src/Valit/ValitRuleBooleanExtensions.cs (hunk 2 above)
Comment (timdeschryver) [reply src/Valit/ValitRuleBooleanExtensions.cs]: 👍 changed it.

**Review by dbarwikowski — APPROVED** [review]

**Review by GooRiOn — APPROVED** [review]

**Comment by GooRiOn** [comment]
Now your code looks great! Welcome aboard buddy ;)

**Comment by timdeschryver** [comment]
@GooRiOn Thanks!  🎉

**Pull request merged**